File: Contracts/Model/Enterprise/Response/DepartmentResponse.cs
namespace Contracts.Model.Enterprise.Response;

public class DepartmentResponse : BaseResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = String.Empty;
    public int Floor { get; set; } 
    public string Description { get; set; } = String.Empty;
    public ICollection<EmployePreviewResponse>? Employees { get; set; }
}

File: ClientApp/src/app/interfaces/enterprise/response/departmentResponse.ts
import { BaseResponse } from "../../baseResponse";
import { EmployePreviewResponse } from "./employePreviewResponse";

export interface DepartmentResponse extends BaseResponse {
    id: number;
    name: string;
    floor: number;
    description: string;
    employees: EmployePreviewResponse[] | null;
}